rdt/hyper_transformer.py

```diff
@@ -382,6 +382,24 @@ def _validate_all_fields_fitted(self):
             warnings.warn('The following fields were specified in the input arguments but not'
                           + f'found in the data: {non_fitted_fields}')
 
+    def _sort_output_columns(self):
+        """Sort ``_output_columns`` according to the ``_input_columns``.
+
+        This method sorts ``_output_columns`` such that the generated output columns
+        have the same order as the original ``_input_columns``.
+        """
+        def _sort_key(output_column):
+            best_i, match_len = float('inf'), 0
+            for i, input_column in enumerate(self._input_columns):
+                if output_column.startswith(input_column):
+                    if match_len < len(input_column):
+                        best_i = i
+                        match_len = len(input_column)
```

Comment:
We have a method called `get_final_output_columns` that returns the output columns derived from a specified input column. Instead of checking that the names match, we can use that to see if it is a match

Reply:
That method doesn't handle combined columns (e.g. `day#month#year.value`). We don't really have any transformers that need it for now, so I'm not sure if we care about it?

Reply:
I don't think we need to worry about that for now